test/unit/watcher.c

```diff
@@ -470,13 +470,16 @@ test_parallel(void)
 }
 
 /**
+ * Checks that all functions are callable and work as usual after
+ * box_watcher_shutdown() except no notification are delivired.
+ *
  * Checks that box_watcher_free() properly unregisters all watchers.
  */
 static void
-test_free(void)
+test_shutdown_and_free(void)
```

Comment:
Please add another test case instead of extending the existing one.

Reply:
Decided to keep this way after discussion f2f yet do some renames.

<details>
<summary>renames diff</summary>

```diff
diff --git a/test/unit/watcher.c b/test/unit/watcher.c
index 4c100ea6b..f92ee71cf 100644
--- a/test/unit/watcher.c
+++ b/test/unit/watcher.c
@@ -476,7 +476,7 @@ test_parallel(void)
  * Checks that box_watcher_free() properly unregisters all watchers.
  */
 static void
-test_shutdown_and_free(void)
+test_free(void)
 {
 	header();
 	plan(17);
@@ -502,22 +502,22 @@ test_shutdown_and_free(void)
 
 	box_watcher_shutdown();
 
-	struct test_watcher wsh1, wsh2;
-	test_watcher_create(&wsh1);
-	test_watcher_create(&wsh2);
+	struct test_watcher w4, w5;
+	test_watcher_create(&w4);
+	test_watcher_create(&w5);
 
-	test_watcher_register(&wsh1, "foo");
-	test_watcher_register_with_flags(&wsh2, "foo", WATCHER_RUN_ASYNC);
-	test_watcher_check_run_count(&wsh1, 0);
-	test_watcher_check_run_count(&wsh2, 0);
+	test_watcher_register(&w4, "foo");
+	test_watcher_register_with_flags(&w5, "foo", WATCHER_RUN_ASYNC);
+	test_watcher_check_run_count(&w4, 0);
+	test_watcher_check_run_count(&w5, 0);
 	test_watch_once("foo", "bar");
 
 	test_broadcast("foo", NULL);
 	test_watcher_check_run_count(&w1, 1);
 	test_watcher_check_run_count(&w2, 1);
 	test_watcher_check_run_count(&w3, 1);
-	test_watcher_check_run_count(&wsh1, 0);
-	test_watcher_check_run_count(&wsh2, 0);
+	test_watcher_check_run_count(&w4, 0);
+	test_watcher_check_run_count(&w5, 0);
 	test_watch_once("foo", NULL);
 
 	test_watcher_resume_sleeping(&w3);
@@ -526,14 +526,14 @@ test_shutdown_and_free(void)
 	test_watcher_check_destroy_count(&w1, 1);
 	test_watcher_check_destroy_count(&w2, 1);
 	test_watcher_check_destroy_count(&w3, 1);
-	test_watcher_check_destroy_count(&wsh1, 1);
-	test_watcher_check_destroy_count(&wsh2, 1);
+	test_watcher_check_destroy_count(&w4, 1);
+	test_watcher_check_destroy_count(&w5, 1);
 
 	test_watcher_destroy(&w1);
 	test_watcher_destroy(&w2);
 	test_watcher_destroy(&w3);
-	test_watcher_destroy(&wsh1);
-	test_watcher_destroy(&wsh2);
+	test_watcher_destroy(&w4);
+	test_watcher_destroy(&w5);
 
 	check_plan();
 	footer();
@@ -582,7 +582,7 @@ main_f(va_list ap)
 	test_ack_unregistered();
 	test_parallel();
 	test_value();
-	test_shutdown_and_free(); /* must be last */
+	test_free(); /* must be last */
 	test_result = check_plan();
 	footer();
 	ev_break(loop(), EVBREAK_ALL);
```

</details>